spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java

```diff
@@ -395,6 +396,15 @@ public JSONObject changeLibraryIdBarcode(HttpSession session, JSONObject json) {
     try {
       User user = securityManager.getUserByLoginName(SecurityContextHolder.getContext().getAuthentication().getName());
       if (!isStringEmptyOrNull(idBarcode)) {
+        Collection<Boxable> previouslyBarcodedItems = requestManager.getBoxablesFromBarcodeList(Arrays.asList(new String[] { idBarcode }));
+        if (!previouslyBarcodedItems.isEmpty()) {
```

Comment:
It should probably check that the id of the current library and the retrieved library are different. `!previouslyBarcodedItems.isEmpty() && !(previouslyBarcodedItems.size() == 1 && previouslyBarcodedItems.get(0).getId() != libraryId)`, where `libraryId` is going to have to get passed in the JSON.